src/plugins/crypto.c

```diff
@@ -563,8 +685,19 @@ static gboolean luks_format (const gchar *device, const gchar *cipher, guint64 k
         }
 #ifdef LIBCRYPTSETUP_2
         else if (luks_version == BD_CRYPTO_LUKS_VERSION_LUKS2) {
+            GError *loc_error = NULL;
             struct crypt_params_luks2 params = ZERO_INIT;
-            params.pbkdf = NULL;
+            struct crypt_pbkdf_type *pbkdf = get_pbkdf_params (extra->pbkdf, &loc_error);
+
+            if (pbkdf == NULL && loc_error != NULL) {
+                crypt_free (cd);
+                g_strfreev (cipher_specs);
+                bd_utils_report_finished (progress_id, loc_error->message);
+                g_clear_error (&loc_error);
```

Comment:
Hmm, would it make sense to propagate `loc_error` to the parent `error` if we're returning `FALSE`?